### DISCUSSION TITLE
Download video

### DISCUSSION THREAD

**Discussion by alex-user-go** [xelaj/mtproto#71] [Show and tell]
Hi, I am trying to download a video from a message. Here what I've got:
```
doc := mediaReal.Document.(*telegram.DocumentObj)
fileLocation := telegram.InputDocumentFileLocation{
	ID: doc.ID,	
	AccessHash:    doc.AccessHash,
	FileReference: doc.FileReference,
	ThumbSize:     "", }
par := telegram.UploadGetFileParams{
	Location:     &fileLocation,
	Limit:        doc.Size}
file, err := client.UploadGetFile(&par)
```
The problem is it gets stuck on `UploadGetFile` forever. Could you show me the right way?

**Comment by quenbyako** [comment]
Sorry for long response, i come back to you, i promise!

**Reply by alex-user-go** [reply]
I found the way to get around it. I put `filelocation` in a channel to read from another gorutine and it works like magic. I would appreciate if you explain why!

**Reply by msadakov** [reply]
How did it work with the channel?

**Comment by msadakov** [comment]
How do I download a file?
